packages/mui-base/src/CheckboxGroup/Root/CheckboxGroupRootContext.ts

```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+import { CheckboxGroupRootContextValue } from './CheckboxGroupRoot.types';
+
+export const CheckboxGroupRootContext = React.createContext<CheckboxGroupRootContextValue | null>(
+  null,
+);
+
+export function useCheckboxGroupRootContext() {
+  const context = React.useContext(CheckboxGroupRootContext);
+  if (context === null) {
+    throw new Error('<CheckboxGroup.Label> must be used within <CheckboxGroup.Root>');
+  }
+  return context;
+}
+
+export function useLooseCheckboxGroupRootContext() {
```

Comment:
Could you use the same pattern as in the Menu (a parameter controlling if the context is required)?